### PR TITLE
fix(discordeno)!: Fix some errors in `DiscordenoConfig`

### PR DESCRIPTION
**Pull request by Fleny113**
1. Fix typo in DesiredProp**rie**tiesBehavior
2. Remove the RecursivePartial<T> from `DiscordenoConfig.desiredProperies.properties` as it makes the object just harder to work with, as it is "lying" since the object should always be constructed with all the props

> [!WARNING]
> This is breaking since it is renaming a type

### CODE REVIEW

File: packages/discordeno/src/bin/config.ts [hunk 1]
```diff
@@ -23,7 +23,7 @@ export const typescriptOptions: ts.CompilerOptions = {
 export function defineConfig(config: RecursivePartial<DiscordenoConfig>): DiscordenoConfig {
   return {
     desiredProperties: {
-      behavior: config.desiredProperties?.behavior ?? DesiredProprietiesBehavior.TypeAsNever,
+      behavior: config.desiredProperties?.behavior ?? DesiredPropertiesBehavior.TypeAsNever,
       properties: createDesiredProprietiesObject(config.desiredProperties?.properties ?? {}),
```
Comment (AwesomeStickz) [inline packages/discordeno/src/bin/config.ts:27]:
```suggestion
      properties: createDesiredPropertiesObject(config.desiredProperties?.properties ?? {}),
```

File: packages/discordeno/src/bin/config.ts [hunk 2]
```diff
@@ -82,7 +82,7 @@ function buildConfig(path: string) {
 
 export interface DiscordenoConfig {
   desiredProperties: {
-    behavior: DesiredProprietiesBehavior
-    properties: RecursivePartial<TransformersDesiredProprieties>
+    behavior: DesiredPropertiesBehavior
+    properties: TransformersDesiredProprieties
```
Comment (AwesomeStickz) [inline packages/discordeno/src/bin/config.ts:86]:
```suggestion
    properties: TransformersDesiredProperties
```

**Review by AwesomeStickz — CHANGES_REQUESTED** [review]

**Comment by AwesomeStickz** [comment]
Could you actually go through entire codebase and replace all proprieties with properties while you’re at it?

**Review by lts20050703 — APPROVED** [review]
LGTM!

**Review by AwesomeStickz — APPROVED** [review]

**Pull request merged**